solvation_analysis/analysis.py

```diff
@@ -0,0 +1,342 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.analysis.rdf import InterRDF
+from MDAnalysis.lib.distances import capped_distance
+import numpy as np
+from solvation_analysis.rdf_parser import identify_solvation_cutoff
+from solvation_analysis.analysis_library import (
+    Coordination,
+    Pairing,
+    Speciation,
+)
+from solvation_analysis.solvation import get_radial_shell, get_closest_n_mol, get_atom_group
+
+
+class Solution(AnalysisBase):
+    """
+    The core class of the solvation module.
+
+    Parameters
+    ----------
+        solute : AtomGroup
```

Comment:
1. No. I just read through the numpy documentation style guide and it was quite helpful for figuring out the right format!
2. I expanded the input to take more diverse kinds of groups. Maybe this gives too much flexibility though? I could also force them to feed in AtomGroups, since it would be at most just calling `OtherGroup.atoms`. What would be better?
3. I changed it to use `MDAnalysis.AtomGroup`, according to the documentation guidelines, that seems to be the "right" way to do it.

Reply:
2 -> I'd say just stick with AtomGroups, it's rare that MDA core supports anything else so users should be quite adapted to it. If folks end up wanting more flexibility you can always add it in later?

Reply:
Makes sense to me!